### ISSUE TITLE
foreach over DataFrame should imply .records?

### ISSUE THREAD

**Issue by jblachly** [blachlylab/koalas#1]
Tried to do `foreach(rec; df)` which felt natural but of course was warned

```
source/main.d(106,13): Error: invalid foreach aggregate agg_df, define opApply(), range primitives, or use .tupleof
```

This can be fixed by foreaching over `df.records`. However, should you build in the range primitives to the DataFrame type and/or forward them to the underlying array for ergonomics?

**Comment by charlesgregory** [comment]
I think is a sound request. I've thought about in the past whether to implement foreach. I also think that potentially ```df.records``` should be private. I can also replicate some pandas functionality/similarity alongside this by providing an ```iterrows``` function.

**Comment by charlesgregory** [comment]
fixed with 4be47ea. Let me know if there are any issues.
